adobe_vipm/flows/sync/agreement.py

```diff
@@ -982,3 +868,332 @@ def _check_adobe_deployment_id(deployment_id: str, adobe_deployment: dict) -> bo
+    """
+    try:
+        agreements = get_agreements_by_3yc_commitment_request_invitation(
+            mpt_client, THREE_YC_TEMP_3YC_STATUSES
+        )
+    except Exception:
+        logger.exception("Unknown exception getting agreements by 3YC enroll status.")
+        raise
+    for agreement in agreements:
+        try:
+            sync_agreement(mpt_client, adobe_client, agreement, dry_run=dry_run, sync_prices=True)
+        except AuthorizationNotFoundError:
+            logger.exception(
+                "AuthorizationNotFoundError synchronizing 3YC enroll status for agreement %s",
+                agreement["id"],
+            )
+        except Exception:
+            logger.exception(
+                "Unknown exception synchronizing 3YC enroll status for agreement %s",
+                agreement["id"],
+            )
+
+
+def sync_all_agreements(mpt_client: MPTClient, adobe_client: AdobeClient, *, dry_run: bool) -> None:
+    """
+    Get all the active agreements to update the prices for them.
+
+    Args:
+        mpt_client: The client used to consume the MPT API.
+        adobe_client: The Adobe API client.
+        dry_run: if True, it just simulate the prices update but doesn't
+        perform it.
+    """
+    agreements = mpt.get_all_agreements(mpt_client)
+    for agreement in agreements:
+        sync_agreement(mpt_client, adobe_client, agreement, dry_run=dry_run, sync_prices=False)
+
+
+def sync_agreement(
+    mpt_client: MPTClient,
+    adobe_client: AdobeClient,
+    agreement: dict,
+    *,
+    dry_run: bool,
+    sync_prices: bool,
+):
+    """
+    Synchronizes a specific agreement with Adobe and MPT clients based on the given parameters.
+
+    Args:
+        mpt_client (MPTClient): Client interface to work with MPT data and services.
+        adobe_client (AdobeClient): Client interface to interact with Adobe services.
+        agreement (dict): A dictionary representing the agreement details to synchronize.
+        dry_run (bool): Flag indicating whether to execute in dry-run mode (no actual changes).
+        sync_prices (bool): Flag indicating whether to synchronize subscription prices.
+    """
+    if agreement["product"]["id"] not in settings.MPT_PRODUCTS_IDS:
+        logger.error("Product %s not in MPT_PRODUCTS_IDS. Skipping.", agreement["product"]["id"])
+        return
+
+    adobe_customer_id = flows_utils.get_adobe_customer_id(agreement)
+    customer = get_customer_or_process_lost_customer(
+        mpt_client, adobe_client, agreement, adobe_customer_id, dry_run=dry_run
+    )
+    if not customer:
+        return
+
+    authorization_id: str = agreement["authorization"]["id"]
+    adobe_subscriptions = adobe_client.get_subscriptions(authorization_id, adobe_customer_id)[
+        "items"
+    ]
+
+    AgreementSyncer(
+        mpt_client, adobe_client, agreement, customer, adobe_subscriptions, dry_run=dry_run
+    ).sync(sync_prices=sync_prices)
+
+
+def get_customer_or_process_lost_customer(
```

Comment:
Umm...it'd be great to split this method in two functions to keep the single responsibility principle. 

On the other hand, It looks like the agreement/sync is not responsible for knowing how to get the customer or process a lost customer. WDYT about creating a customer class / file to handle it?